### DISCUSSION TITLE
Support Forms in Chat

### DISCUSSION THREAD

**Discussion by totalhack** [danny-avila/LibreChat#1371] [Feature Requests & Suggestions]
Have you considered a feature where certain responses from the LLM might be translated to a form-like UI instead of plaintext input?

Simplest example: if the LLM asks the user a yes/no question, return a yes and a no button. The same logic could be extended to date inputs, cases where a dropdown might be appropriate, etc. 

Beyond those obvious cases, I imagine many business use cases where you want the LLM to collect more complex input to complete a task, such as forming an analytics query for an internal tool.

**Comment by danny-avila** [comment]
Hey @totalhack sorry I didn’t get to reply sooner

Based on what you said, the first thing that comes to mind is AIPRM-like prompts, where you can create form-like variables for the prompt. Is that pretty accurate to what you mean? We can definitely explore this through prompts when they are integrated.

**Reply by danny-avila** [reply]
The AIPRM extension for ChatGPT also has custom “continue” actions where you can customize ready to go prewritten responses, I would be open to seeing that in this project also but won’t be a priority at the moment

**Reply by totalhack** [reply]
I'm not familiar with AIPRM but after a quick Google it doesn't quite look like what I am talking about. My suggestion is about using more user-friendly UI elements in the chat when possible. For inputs, you might support buttons, radios, dropdowns, datepickers, etc. For outputs, charts, data tables, etc. 

As for how these things are decided / triggered that's all TBD. The simplest example is if the model needs to ask the user a yes no question, perhaps for confirmation of some action, then that should be two buttons instead of the user having to type out "yes" or "no". Random thought...maybe this could be some extension of function calling, where the model can infer what function arguments it doesn't have yet and ask the user via inline form elements. Maybe there is a better/cleaner way but ideally it would be extendible and flexible.

**Reply by secondtruth** [reply]
@totalhack I thought about this too. It could be very useful for many tasks.
